demos/segmentation_demo/cpp/main.cpp

```diff
@@ -170,7 +170,7 @@ cv::Mat applyColorMap(cv::Mat input) {
     return out;
 }
 
-cv::Mat renderSegmentationData(const SegmentationResult& result) {
+cv::Mat renderSegmentationData(const ImageProcessingResult& result) {
```

Comment:
I really don't like naming. If we want to use same result class both for segmentation and image processing (that make sense) we need some neutral name for it, not bound to particular demo name and suitable for both. For instance "ImageResult", "MatResult" or any other. If we won't come up to good name, please keep separate result classes, or make #typedef SegmentationResult ImageProcessingResult

Reply:
Now I using `ImageResult` name.